apps/designer/app/designer/features/style-panel/shared/css-value-input/css-value-input.tsx

```diff
@@ -59,7 +58,7 @@ const useHandleOnChange = (
     }
 
     // We want to switch to unit mode if entire input is a number.
-    if (isNumericString(input)) {
+    if (isNumericString(input) && isValid(property, input + "px")) {
```

Comment:
What if its not valid with current unit `unit = valueRef.current.type === "unit" ? valueRef.current.unit : "px"`
Here we validate with fallback and later call onChange with other unit

Reply:
Also would be good to have some explanation why px is fallback unit

Reply:
Also we have `toValue` https://github.com/webstudio-is/webstudio-designer/blob/0b37ccdefe7dcf692a5efe5eba33a51adc47ec71/packages/css-engine/src/core/to-value.ts#L12